### PR TITLE
Support for multiple resource groups

### LINKED ISSUE #294: Provide support for scraping multiple resource groups
Currently the metrics configuration is as follows:

```
azureMetadata:
  tenantId: c8819874-9e56-4e3f-b1a8-1c0325138f27
  subscriptionId: 0f9d7fea-99e8-4768-8672-06a28514f77e
  resourceGroupName: promitor
metricDefaults:
  aggregation:
    interval: 00:05:00
metrics:
  - name: demo_generic_queue_size
    description: "Amount of active messages of the 'myqueue' queue (determined with Generic provider)"
    resourceType: Generic
    resourceUri: Microsoft.ServiceBus/namespaces/promitor-messaging
    filter: EntityName eq 'orders'
    azureMetricConfiguration:
      metricName: ActiveMessages
      aggregation:
        type: Total
```

This works for the scenario of monitoring a single set of metrics within one resource group, however best practices dictates that not all resources are located within a single resource group within a subscription.  Moving the resourceGroupName to the metrics configuration would provide support for metrics from multiple resource groups within Azure.

Example configuration:

```
azureMetadata:
  tenantId: c8819874-9e56-4e3f-b1a8-1c0325138f27
  subscriptionId: 0f9d7fea-99e8-4768-8672-06a28514f77e
metricDefaults:
  aggregation:
    interval: 00:05:00
metrics:
  - name: demo_generic_queue_size_rg1
    description: "Amount of active messages of the 'myqueue' queue (determined with Generic provider)"
    resourceType: Generic
    resourceGroupName: RG_1
    resourceUri: Microsoft.ServiceBus/namespaces/promitor-messaging
    filter: EntityName eq 'orders'
    azureMetricConfiguration:
      metricName: ActiveMessages
      aggregation:
        type: Total
  - name: demo_generic_queue_size_rg2
    description: "Amount of active messages of the 'myqueue' queue (determined with Generic provider)"
    resourceType: Generic
    resourceGroupName: RG_2
    resourceUri: Microsoft.ServiceBus/namespaces/promitor-messaging
    filter: EntityName eq 'orders'
    azureMetricConfiguration:
      metricName: ActiveMessages
      aggregation:
        type: Total
```

### PR DESCRIPTION
**Pull request by brusMX**
Fixes #294 

## Proposed Changes

  - Refactoring Scraping to be done per Resource Group per metric

### CODE REVIEW

File: src/Promitor.Core.Scraping/Configuration/Model/Metrics/MetricDefinition.cs [hunk 1]
```diff
@@ -17,6 +17,11 @@ public abstract class MetricDefinition
         /// </summary>
         public string Name { get; set; }
 
+        /// <summary>
+        ///     Override the Resource Group name at the metric level (to enable multiple RG)
```
Comment (tomkerkhove) [inline src/Promitor.Core.Scraping/Configuration/Model/Metrics/MetricDefinition.cs]: Let's try to be more descriptive here

```suggestion
        ///     Specify a resource group to scrape that defers from the default resource group.
        ///     This enables you to do multi-resource group scraping with one configuration file.
```

File: src/Promitor.Core.Scraping/Scraper.cs [hunk 2]
```diff
@@ -15,109 +15,110 @@
 
 namespace Promitor.Core.Scraping
 {
+  /// <summary>
+  ///     Azure Monitor Scrape
```
Comment (tomkerkhove) [inline src/Promitor.Core.Scraping/Scraper.cs]: Let's fix this typo
```suggestion
  ///     Azure Monitor Scraper
```

File: src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithAzureStorageQueueYamlSerializationTests.cs [hunk 3]
```diff
@@ -11,71 +11,75 @@
...
+      var bogusGenerator = new Faker<StorageQueueMetricDefinition>()
+          .StrictMode(ensureRulesForAllProperties: true)
+          .RuleFor(metricDefinition => metricDefinition.Name, faker => faker.Name.FirstName())
+          .RuleFor(metricDefinition => metricDefinition.Description, faker => faker.Lorem.Sentence(wordCount: 6))
+          .RuleFor(metricDefinition => metricDefinition.ResourceType, faker => ResourceType.StorageQueue)
+          .RuleFor(metricDefinition => metricDefinition.AccountName, faker => faker.Name.LastName())
+          .RuleFor(metricDefinition => metricDefinition.QueueName, faker => faker.Name.FirstName())
+          .RuleFor(metricDefinition => metricDefinition.SasToken, faker => $"?sig={Base64UrlEncoder.Encode(faker.Lorem.Sentence(wordCount: 3))}")
+          .RuleFor(metricDefinition => metricDefinition.AzureMetricConfiguration, faker => bogusAzureMetricConfiguration)
+          .RuleFor(metricDefinition => metricDefinition.ResourceGroupName, faker => resourceGroupName)
```
Comment (tomkerkhove) [inline src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithAzureStorageQueueYamlSerializationTests.cs]: Please add this to our generic one as well

File: docs/configuration/metrics/index.md [hunk 4]
```diff
@@ -41,18 +53,23 @@ metrics:
 - `metricDefaults.aggregation.interval` - The default interval which defines over what period measurements of a metric should be aggregated.
 
 ## Metrics
+
 Every metric that is being declared needs to define the following fields:
+
 - `name` - Name of the metric that will be exposed in the scrape endpoint for Prometheus
 - `description` - Description for the metric that will be exposed in the scrape endpoint for Prometheus
 - `resourceType` - Defines what type of resource needs to be queried.
+- `resourceGroupName` - (Optional)  This field is used to override the resource group declared in `azureMetadata`. This allows Promitor to use multiple resource groups in one single yaml file. Provide one resource group per metric.
```
Comment (tomkerkhove) [inline docs/configuration/metrics/index.md]: Personally find it a bit too technical - How about this?

```suggestion
- `resourceGroupName` - _(Optional)_  Capability to specify a resource group to scrape for this metric. This allows you to specify a different resource group from the one configured in `azureMetadata`. This allows you to scrape multiple resource groups with one single configuration.
```

I left out `Provide one resource group per metric.` as I'm not sure what you mean here, that you need to configure it at least on one place either metric or azure metadata? Isn't this already covered given the one in azure metadata is not optional?

Thoughts?

File: src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithAzureStorageQueueYamlSerializationTests.cs (hunk 3 above)
Comment (brusMX) [reply src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithAzureStorageQueueYamlSerializationTests.cs]: There is not a test for `MetricsDeclarationWithGenericAzureYamlSerializationTests.cs` inside of `Tests/Serialization/MetricsDeclaration/`. Let's do a new issue for this

Comment (tomkerkhove) [reply src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithAzureStorageQueueYamlSerializationTests.cs]: :+1:

File: docs/configuration/metrics/index.md (hunk 4 above)
Comment (tomkerkhove) [reply docs/configuration/metrics/index.md]: @brusMX 	can you redo this one as well please?

**Pull request merged**